daemon/cmd/daemon_main.go

```diff
@@ -1764,6 +1764,25 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 		}
 	}
 
+	if option.Config.EnableEnvoyConfig {
```

Comment:
Why is this being added to `daemon/cmd`? `pkg/envoy` already is a module. This should be in `pkg/envoy` as part of `envoy.Cell`.

Reply:
This option is not added in this PR. Refactoring the old code is beyond the scope of this PR.

Reply:
At least one tricky thing here is the ingress IP allocations that happen earlier here. I'm asking too much here as part of this PR! I think once ingress IP allocation is somewhere outside then we can model this also outside daemon/cmd.